packages/loader/container-loader/src/container.ts

```diff
@@ -946,6 +946,79 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
     }
 
+    public connect() {
+        if (!this.closed && !this.connected) {
+            // Note: no need to fetch ops as we do it preemptively as part of DeltaManager.attachOpHandler().
+            // If there is gap, we will learn about it once connected, but the gap should be small (if any),
+            // assuming that connect() is called quickly after initial container boot.
+            this.connectInternal({ reason: "DocumentOpenResume", fetchOpsFromStorage: false });
+        }
+    }
+
+    private connectInternal(args: IConnectionArgs) {
+        assert(!this.closed, "Attempting to connect() a closed DeltaManager");
+        assert(this._attachState === AttachState.Attached, "Attempting to connect() a container that is not attached");
+
+        // Resume processing ops
+        if (!this.resumedOpProcessingAfterLoad) {
+            this.resumedOpProcessingAfterLoad = true;
+            this._deltaManager.inbound.resume();
+            this._deltaManager.inboundSignal.resume();
+        }
+
+        // Ensure connection to web socket
+        this.connectToDeltaStream(args);
+
+        const mode = ReconnectMode.Enabled;
+        const currentMode = this._deltaManager.connectionManager.reconnectMode;
+
+        // Set Auto Reconnect Mode
+        if (currentMode !== mode) {
+            const now = performance.now();
+            const duration = now - this.setAutoReconnectTime;
+            this.setAutoReconnectTime = now;
+
+            this.mc.logger.sendTelemetryEvent({
+                eventName: "AutoReconnectEnabled",
+                connectionMode: this.connectionMode,
+                connectionState: ConnectionState[this.connectionState],
+                duration,
+            });
+
+            this._deltaManager.connectionManager.setAutoReconnect(mode);
+        }
+    }
+
+    public disconnect() {
+        if (!this.closed && this.connectionState !== ConnectionState.Disconnected) {
```

Comment:
Note that if we are in the process of attempting to connect, then this.connectionState would be disconnected, and calling this API would be noop. But the outcome - connection will succeed eventually, while user clearly does not want that. This is a new flow that did not exist before, thus I think we do not have good mechanisms to control it (i.e. there is no way to cancel pending connection, the only thing we do is to throw away connection on success if container is closed).

I think this needs to be redesigned in some way and implemented correctly (i.e. doing what users expect from this API). It's probably fine to do it as separate PR, but we should do it right after that, as it is not that uncommon situation (especially when user is offline).

Reply:
If you allow to go past that check and assess this._deltaManager.connectionManager.reconnectMode in disconnectInternal(), then I think it will do 95% of what is required (assuming that DeltaManager can throw this connection away once it connects and it realizes that reconnect mode is Disabled, though I think we need to use Never, not Disabled, and review code around it, as I think some asserts will not like it)

Reply:
Would you be able to explain why `Never` is preferred over `Disabled`? At the moment we seem to only use `Never` to indicate that a container should never be able to reconnect in the future. This doesn't seem to align with what we want to accomplish with `disconnect()`, since we want to allow the container to reconnect at a later point when calling `connect()`.

Reply:
I think Disabled is used (for example) for agents (including summarizer) and it allows one connection. So decision not to reconnect is made on loss of connection, not when connection is successfully established. Here, when user wants to stop connectivity flow, it's too late to check (pending connection is already in flight), and once pending connection is established it can't make decision to abandon such connection for Disalbed state (as that will kill agent connection flow). So we need some other state to differentiate "first connection" vs. "connection should not be allows".
I think Never might also be wrong (as some other things are tied to it), so more exploration is required to properly handle that case.

Reply:
Sounds like it's probably best to look into this (and the other concerns you mentioned above with `disconnect()` during a pending connection) in a separate PR like you suggested.